llvm/lib/Transforms/Utils/LowerMemIntrinsics.cpp

```diff
@@ -456,6 +456,48 @@ static void createMemMoveLoop(Instruction *InsertBefore, Value *SrcAddr,
   ElseTerm->eraseFromParent();
 }
 
+static void createMemSetPatternLoop(Instruction *InsertBefore, Value *DstAddr,
```

Comment:
This code looks identical to createMemSetLoop to my eye, is there a reason this can't be shared?